### PR TITLE
[VM] check DLManagedTensor for conditions to construct NDArray

### PR DESCRIPTION
**Pull request by vvchernov**
During developing #11003 #11391 It was observed that DLTensor should satisfy some conditions (data contiguous and alignment) for correct construction of NDArray based on it to avoid problems with memory using.
Further analysis of NDArray class methods showed the same problem in other public method. There is fix of it.

@tkonolige could you check it?

### CODE REVIEW

**Review by tkonolige — APPROVED** [review]
Thanks @vvchernov!

File: src/runtime/ndarray.cc [hunk 1]
```diff
@@ -240,6 +240,10 @@ NDArray NDArray::FromDLPack(DLManagedTensor* tensor) {
   data->SetDeleter(Internal::DLPackDeleter);
   // fill up content.
   data->manager_ctx = tensor;
+  ICHECK(::tvm::runtime::IsContiguous(tensor->dl_tensor))
+      << "DLManagedTensor is not contiguous. It does not support for now";
```
[the remark this answers is not in the document]
Comment (vvchernov) [reply src/runtime/ndarray.cc:244]: Ok, I will redo it. But I based on similar description in other ICHECK. I think TVM design assumes contiguous tensors only and what we can potentially support in the future it is copying from incontiguous to contiguous ones.

[the remark this answers is not in the document]
Comment (tkonolige) [reply src/runtime/ndarray.cc]: Personally I think saying something will be supported in the future doesn't really add any useful information. But, if you do want to say that "Only contiguous DLManagedTensors are currently supported." would be a better message.

[the remark this answers is not in the document]
Comment (vvchernov) [reply src/runtime/ndarray.cc]: Hello @tkonolige! Please see again

[the remark this answers is not in the document]
Comment (tkonolige) [reply src/runtime/ndarray.cc]: How about "DLTensor is not contiguous. Copying from non-contiguous data is currently not supported."

[the remark this answers is not in the document]
Comment (tkonolige) [reply src/runtime/ndarray.cc]: The wording on those lines is a little awkward but I believe they mean that ArrayCopyFromBytes and ArrayCopyToBytes only support contiguous data.

**Review by tmoreau89 — APPROVED** [review]
Thanks @tkonolige for the review, LGTM

**Pull request merged**